src/analyses/goto_check.cpp

```diff
@@ -909,7 +909,7 @@ void goto_checkt::pointer_validity_check(
   const exprt &access_ub,
   const irep_idt &mode)
 {
-  if(mode!=ID_java && !enable_pointer_check)
+  if(!enable_pointer_check)
```

Comment:
Yes: if for some reason you want to use goto-check on Java code (probably in this case you have omitted or customised java-bytecode-instrument), the checks below that point are still inapplicable to Java.